packages/agent-toolkit/src/core/tools/platform-api-tools/create-item-tool/create-item-tool.ts

```diff
@@ -0,0 +1,149 @@
+  }
+
+  getInputSchema(): CreateItemToolInput {
+    if (this.context?.boardId) {
+      return createItemToolSchema;
+    }
+
+    return createItemInBoardToolSchema;
+  }
+
+  protected async executeInternal(input: ToolInputType<CreateItemToolInput>): Promise<ToolOutputType<never>> {
+    const boardId = this.context?.boardId ?? (input as ToolInputType<typeof createItemInBoardToolSchema>).boardId;
+
+    // Check for conflicting parameters
+    if (input.duplicateFromItemId && input.parentItemId) {
+      throw new Error('Cannot specify both parentItemId and duplicateFromItemId. Please provide only one of these parameters.');
+    }
+
+    //three paths: duplicate item, create subitem, or create regular item
+    if (input.duplicateFromItemId) {
+      return await this.duplicateAndUpdateItem(input, boardId);
+    } else if (input.parentItemId) {
+      return await this.createSubitem(input);
+    } else {
+      return await this.createNewItem(input, boardId);
+    }
+  }
+
+  private async duplicateAndUpdateItem(input: ToolInputType<CreateItemToolInput>, boardId: number): Promise<ToolOutputType<never>> {
+
+    const duplicateVariables = {
+      boardId: boardId.toString(),
+      itemId: input.duplicateFromItemId!.toString()
+    };
+    
+    const duplicateRes = await this.mondayApi.request<DuplicateItemMutation>(duplicateItem, duplicateVariables);
+    
+    if (!duplicateRes.duplicate_item?.id) {
+      throw new Error('Failed to duplicate item');
+    }
+
+    let columnValuesParsed;
+    try {
+      columnValuesParsed = JSON.parse(input.columnValues);
+    } catch (error) {
+      throw new Error('Invalid JSON in columnValues');
+    }
+    
+    const columnValuesAndName = {
+      ...columnValuesParsed,
+      name: input.name,
+    };
+
+    // Now update the duplicated item with the new name and column values
+    const changeColumnValuesTool = new ChangeItemColumnValuesTool(this.mondayApi, this.apiToken, { boardId: boardId });
+    const updateRes = await changeColumnValuesTool.execute({
+      itemId: parseInt(duplicateRes.duplicate_item.id),
+      columnValues: JSON.stringify(columnValuesAndName),
+    });
+
+    if (updateRes.content.includes('Error')) {
+      throw new Error('Failed to update duplicated item: ' + updateRes.content);
+    }
+
+    return {
+      content: `Item ${duplicateRes.duplicate_item.id} successfully duplicated from ${input.duplicateFromItemId} and updated`,
+    };
+  }
+
+  private async createSubitem(input: ToolInputType<CreateItemToolInput>): Promise<ToolOutputType<never>> {
+    // Create subitem using the create_subitem mutation
+    const variables = {
+      parentItemId: input.parentItemId!.toString(),
+      itemName: input.name,
+      columnValues: input.columnValues,
+    };
+
+    const res = await this.mondayApi.request<CreateSubitemMutation>(createSubitem, variables);
+
+    if (!res.create_subitem?.id) {
+      throw new Error('Failed to create subitem');
```

Comment:
If we don't return the original error, the agent won't know what to do, we should add action item for each error 🙏